test/src/mbgl/test/http_server.cpp

```diff
@@ -0,0 +1,197 @@
+
+    server->Get("/not-found", [](const Request&, Response& res) {
+        res.status = 404;
+        res.set_content("Not Found!", "text/plain");
+    });
+
+    server->Get("/permanent-error", [](const Request&, Response& res) {
+        res.status = 500;
+        res.set_content("Server Error!", "text/plain");
+    });
+
+    std::atomic_int temporaryErrorCounter(0);
+    server->Get("/temporary-error", [&](const Request&, Response& res) {
+        if (temporaryErrorCounter == 0) {
+            res.status = 500;
+        } else {
+            res.status = 200;
+            res.set_content("Hello World!", "text/plain");
+        }
+        temporaryErrorCounter++;
+    });
+
+    server->Get("/rate-limit", [](const Request& req, Response& res) {
+        if (req.has_param("std")) {
+            res.set_header("Retry-After", "1");
+        } else if (req.has_param("mbx")) {
+            res.set_header("x-rate-limit-reset", std::to_string(util::now().time_since_epoch().count() + 1));
+        }
+        res.status = 429;
+    });
+
+    std::atomic_bool styleFailOnce500(true);
+    server->Get("/style-fail-once-500", [&](const Request&, Response& res) {
+        if (styleFailOnce500) {
+            res.status = 500;
+            res.set_content("Not found!", "text/plain");
+            styleFailOnce500 = false;
+        } else {
+            res.status = 200;
+            res.set_content(R"({ "version": 8, "name": "Test Style" })", "text/plain");
+        }
+    });
+
+    std::atomic_bool styleFailOnce404(true);
+    server->Get("/style-fail-once-404", [&](const Request&, Response& res) {
+        if (styleFailOnce404) {
+            res.status = 404;
+            res.set_content("Not found!", "text/plain");
+            styleFailOnce404 = false;
+        } else {
+            res.status = 200;
+            res.set_content(R"({ "version": 8, "name": "Test Style" })", "text/plain");
+        }
+    });
+
+    std::atomic_bool styleFailOnce404Cache(true);
+    server->Get("/style-fail-once-404-cache", [&](const Request&, Response& res) {
+        if (styleFailOnce404Cache) {
+            res.set_header("Cache-Control", "max-age=30");
+            res.status = 404;
+            res.set_content("Not found!", "text/plain");
+            styleFailOnce404Cache = false;
+        } else {
+            res.status = 200;
+            res.set_content(R"({ "version": 8, "name": "Teste Style" })", "text/plain");
+        }
+    });
+
+    server->Get("/delayed", [](const Request&, Response& res) {
+        usleep(200000);
+        res.status = 200;
+        res.set_content("Response", "text/plain");
+    });
+
+    server->Get(R"(/load/(\d+))", [](const Request req, Response& res) {
+        auto numbers = req.matches[1];
+        res.set_content("Request " + std::string(numbers), "text/plain");
+    });
+
+    server->listen("127.0.0.1", 3000);
+}
+
+HttpServer::HttpServer() {
+    server = std::make_unique<httplib::Server>();
```

Comment:
Quickly checked httplib and it has `CPPHTTPLIB_THREAD_POOL_COUNT` option that might be useful to reduce number of threads?

Reply:
Very good point, will limit the nbr of threads.